### ISSUE TITLE
Multiline superclasses regression in 0.40

### ISSUE THREAD

**Issue by ZacSweers** [pinterest/ktlint#1046]
## Expected Behavior
This should format consistently

## Observed Behavior
It fails on its own formatting

## Steps to Reproduce
```kotlin
const val MY_LONG_CONSTANT = 1
interface SomeInterface
interface AnotherInterface
abstract class BaseClass(textColor: Int, bgColor: Int, thirdArg: Int)

class SubClass private constructor(
  private val textColor: Int,
  private val bgColor: Int
) : BaseClass(
  textColor,
  bgColor,
  MY_LONG_CONSTANT
),
  SomeInterface,
  AnotherInterface {
}
```

When formatting it, it will result in this:

```kotlin
class SubClass private constructor(
  private val textColor: Int,
  private val bgColor: Int
) : BaseClass(
    textColor,
    bgColor,
    MY_LONG_CONSTANT
  ),
  SomeInterface,
  AnotherInterface {
}
```

But then complain that the indentation is unexpected

```
ChannelTagSpan.kt:19:5: Unexpected indentation (expected 2, actual 4)
```

This is a regression from 0.39.0

## Your Environment

This is our editorconfig

```
[*.{kt,kts}]
indent_size=2
continuation_indent_size=2
disabled_rules=final-newline
```

**Comment by paul-dingemans** [comment]
In this issue the ArgumentListWrappingRule and IndentRule are both triggered at the same lines and each rule reports errors found. The full lint report for this issues is:
```
issue1046.kt:7:1: Unexpected indentation (4) (should be 2)
issue1046.kt:7:5: Unexpected indentation (expected 2, actual 4)
issue1046.kt:8:1: Unexpected indentation (4) (should be 2)
issue1046.kt:8:5: Unexpected indentation (expected 2, actual 4)
issue1046.kt:10:1: Unexpected indentation (4) (should be 2)
issue1046.kt:11:1: Unexpected indentation (4) (should be 2)
issue1046.kt:12:1: Unexpected indentation (4) (should be 2)
issue1046.kt:14:1: Unexpected indentation (4) (should be 2)
issue1046.kt:15:1: Unexpected indentation (4) (should be 2)
issue1046.kt:15:22: Unnecessary block ("{}")
```

It is confusing that the style of messages printed by those rules is different. But both do report that the actual indentation is 4 while 2 was expected. At least the messages should be aligned.

An interesting question is whether we should attempt to avoid duplicate messages although they are cause by different rules. Or should we always report the rule in which the violation was found like the verbose mode does. @romtsn Do you have an opinion about that?

**Comment by t-kameyama** [comment]
I think this issue has been resolved by https://github.com/pinterest/ktlint/pull/1006.

**Comment by paul-dingemans** [comment]
Issue can indeed *not* be reproduced anymore with current master.

**Comment by romtsn** [comment]
@ZacSweers we'll try to make a 0.41.0 release this month, you could give the snapshot a try if you will.
